alpha/lib/model/KuserKgroupPeer.php

```diff
@@ -99,7 +99,36 @@ public static function retrieveKgroupIdsByKuserId($kuserId){
 
 		return self::$kgroupIdsByKuserId[$kuserId];
 	}
-	
+
+	/**
+	 * @param $kuserId
+	 * @param $partnerId
+	 * @return array|mixed
+	 */
+	public static function retrieveKgroupIdsByKuserIdAndPartnerId($kuserId, $partnerId)
+	{
+		if (isset(self::$kgroupIdsByKuserId[$kuserId])){
+			return self::$kgroupIdsByKuserId[$kuserId];
+		}
+
+		$c = new Criteria();
+		$c->add(KuserKgroupPeer::KUSER_ID, array($kuserId), Criteria::IN);
+		$c->addAnd ( KuserKgroupPeer::STATUS, array(KuserKgroupStatus::DELETED), Criteria::NOT_IN);
```

Comment:
These 2 rules are not needed, you already get them by the default criteria , (Status != delete and partner ID).
And also remove the call to  setUseCriteriaFilter(false)